src/examples/libpmem2/aof.c

```diff
@@ -0,0 +1,417 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2020, Intel Corporation */
+
+/*
+ * aof.c -- "Persistently-correct" implementation of an append-only file
+ * structure. The user may do single/multiple appends and flush them all.
+ * When an append/appendv command is done, the application persists all append
+ * data, updates the header (especially information about the end of the valid
+ * data in the file), and persists the changes done to the header.
+ */
+
+#include <errno.h>
+#include <ex_common.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <libpmem2.h>
+
+#define AOF_HDR_SIGNATURE "PMEM2_AOF"
+#define AOF_HDR_SIGNATURE_LEN (sizeof(AOF_HDR_SIGNATURE))
+
+#define MIN_SIZE (1 << 20) /* 1 MiB */
+
+/*
+ * aof_header -- AOF header
+ */
+struct aof_header {
+	uint8_t signature[AOF_HDR_SIGNATURE_LEN];
+	uint64_t end_offset;
+};
+
+/*
+ * aof -- append-only file application context
+ */
+struct aof {
+	struct aof_header *header;
+	struct pmem2_map *map;
+	void *end_ptr;
+	void *tail;
```

Comment:
the purpose of these 2 fields is not very clear - I think naming of them could be improved, but I don't have a suggestion yet